### DISCUSSION TITLE
How to add controller to Deck.gl component?

### DISCUSSION THREAD

**Discussion by mmz15** [visgl/deck.gl#5516] [Q&A] [answered]
I'm trying to disable `scrollZoom` in my `DeckGL` component. I am, however, having difficulty understanding the [documentation](https://deck.gl/docs/api-reference/core/map-controller) for passing Controllers to these objects. My current implementation simplified for brevity is: 

```
  new Deck({
    controller: { scrollZoom: false },
    initialViewState: {
      latitude: 42.508108,
      longitude: -71.508701,
      zoom: 10,
      maxZoom: 18,
      minZoom: 4,
      pitch: 45,
      bearing: 0,
    },
  });

return (
    <>
      <DeckGL
        ContextProvider={MapContext.Provider}
        controller={true}
        effects={effects}
        getTooltip={getTooltip}
        initialViewState={INITIAL_VIEW_STATE_AREA}
        layers={layers}
        onWebGLInitialized={onInitialized}
      >
        <InteractiveMap
          reuseMaps
          ref={mapRef}
          mapStyle={MAP_STYLE}
          preventStyleDiffing={true}
          mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
        />
      </DeckGL>
    </>
```

The code above returns a static map. I would've expected the code map to enable you to pan but not scroll.

Is there an example or additional documentation that provides a better understanding of this implementation?

**Comment by Pessimistress** [comment, marked as answer]
`Deck` is the non-React flavor of `DeckGL`. To adapt the example in the documentation to your use case, just pass the `controller` prop to your DeckGL component.

```
<DeckGL controller={{scrollZoom: false}} ... />
```

**Reply by mmz15** [reply]
Works like a charm. Appreciate the clarification!
